### PR TITLE
net/wireless: add support for netdev private ioctl

### PR DESCRIPTION
**Pull request by anchao**
## Summary

1. net/wireless: add support for netdev private ioctl

2. net/wireless: separate command catagory of bt/pktradio from wireless ioctl

```
1. separate command catagory of bt/pktradio from wireless ioctl
2. Remove commoand count limit to support vendor command
```

## Impact

N/A

## Testing

CI check

### CODE REVIEW

File: include/nuttx/wireless/pktradio.h [hunk 1]
```diff
@@ -37,33 +37,29 @@
 
 /* Packet radio network device IOCTL commands. */
 
-#ifndef WL_NPKTRADIOCMDS != 3
```
[the remark this answers is not in the document]
Comment (anchao) [reply include/nuttx/wireless/pktradio.h:40]: Done

**Review by xiaoxiang781216 — APPROVED** [review]

File: include/nuttx/wireless/wireless.h [hunk 2]
```diff
@@ -150,6 +150,42 @@
...
+/* Previously, we were using SIOCDEVPRIVATE, but we now have our
+ * separate range because of collisions with other tools such as
+ * 'mii-tool'.
+ * We now have 32 commands, so a bit more space ;-).
+ * Also, all 'even' commands are only usable by root and don't return the
+ * content of ifr/iwr to user (but you are not obliged to use the set/get
+ * convention, just use every other two command). More details in iwpriv.c.
+ * And I repeat : you are not forced to use them with iwpriv, but you
+ * must be compliant with it.
+ */
```
Comment (gustavonihei) [inline include/nuttx/wireless/wireless.h:173]: This comment does not seem to be pertinent to this project, considering the references to `mii-tool` and `iwpriv.c`.

Comment (anchao) [reply include/nuttx/wireless/wireless.h:173]: Yes, let me remove the comment

**Review by gustavonihei — APPROVED** [review]
LGTM

**Pull request merged**